### ISSUE TITLE
Future ideas for speed limit tool (scope creen from PR 1168)

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#1221]
Future idea scope creep gathered from PR #1168 : 

* Mouse wheel:
    * when in speed window, could cycle the selected speed
    * when over a sign in overlay, could adjust speed of that specific sign?
* Icon overlay fade - only when nearing max distance from cam?
* Disable speed shortcuts (mod option) - possibly limit to disabling the `Alt`+`0..9` ones?
* Treat `Ø` as a speed:
    * Move button to end of speed bar (eg. after 90mph) so it's logical progression 0 -> 90 -> unlimited
    * Right align under-button hint text for that button
    * Have it change size in same way the other speed buttons do
    * Include in speed cycling shortcuts (could possibly ditch the `/`,`\` shortcuts?)

**Comment by aubergine10** [comment]
Speed overlay signs seem bigger in lane mode than they are in segment mode.

**Comment by aubergine10** [comment]
Find It mod - issue regarding it trapping `Alt`+`1..0` shortcuts - https://github.com/sway2020/FindIt2/issues/135

**Comment by aubergine10** [comment]
very low pri: Allow dragging of speed window by the bottom (button hint area)?

**Comment by aubergine10** [comment]
If speed window loses focus the key shortcuts stop working again - steps to reproduce:

1. Open speed tool - shortcuts work fine
2. Click `(?)` on TMPE main toolbar - speed shortcuts stop working (until a button on speed bar is clicked)

**Comment by aubergine10** [comment]
Remove circle backgrounds from buttons (seg vs. lane, and defaults) at side of speed window?

**Comment by aubergine10** [comment]
Allow setting defaults per-lane? Probably a huge change tho...

Example use cases:

* Setting slower outer lanes on large roads to promote traffic using faster inner lanes
* Setting slow 'local route' lanes on roads with multiple medians (eg. https://github.com/CitiesSkylinesMods/TMPE/issues/1220#issuecomment-987500836 )

**Comment by aubergine10** [comment]
OSD - Merge Increase/Decrease shortcuts on to one line, so the Alt shortcuts can be moved to it's own line ?
![image](https://user-images.githubusercontent.com/1386719/144964620-3106d797-c0a3-47c7-8e4c-80dcf855e127.png)

**Comment by aubergine10** [comment]
Find It [v2.7.3-beta1](https://steamcommunity.com/sharedfiles/filedetails/changelog/2187834843) now disables its secondary shortcuts (Alt+0..9) when the Find It toolbar is not open:

> - Secondary keyboard shortcuts are now disabled by default when Find It 2 is closed  
> These shortcuts (Alt + 0-9/V) will only function when Find it 2's panel is opened  
> Main keyboard shortcut (Ctrl + F) is unaffected  

https://github.com/sway2020/FindIt2/issues/135

**Comment by aubergine10** [comment]
https://github.com/CitiesSkylinesMods/TMPE/issues/925

**Comment by aubergine10** [comment]
Did this get included: https://github.com/CitiesSkylinesMods/TMPE/issues/988

**Comment by kvakvs** [comment]
> Did this get included: #988

i believe this was quick-fixed soon after report

**Comment by aubergine10** [comment]
From https://github.com/CitiesSkylinesMods/TMPE/pull/1215#issuecomment-989521250 : 

> How much scope is there for UI change in this PR? I'm finding the current way it works to be a bit... clunky, particularly as the mph vs. km/h checkbox can change based on which theme is chosen, and when clicking the in-game mph / km/h button I can sometimes get an unexpected theme and then end up having to go in to mod options anyway = ugh.
>  
> Would it be better to have the theme selection done entirely via the in-game UI, for example:
>  
> * Open speeds tool: If first use (or config theme id not found / invalid) switch UI to "select theme" mode:
>     * Window title bar text: "Select speed icon theme"
>     * Speed bar gets replaced with an icon theme bar - eg. `30.png` icon from each theme, arranged in a row
>     * _Alternatively, mph themes could show `30.png` and km/h themes could show `50.png`_ = easy visual cue to the speed unit
>     * _If, at later date, there's too many themes, we can add scroll buttons or more rows or whatever_
>     * Over/Under-button hint text shows (top) mph or km/h and (bottom) country
>     * _Countries with both mph and km/h would get two entries in the theme bar_
>     * User clicks desired icon - that theme is loaded and UI switches to speed bar mode
> * In speed bar mode, clicking the mph / km/h button (it's tooltip could be smth like "Select speed icon theme") goes to select theme mode
>  
> This way user is prompted to set theme on first use of tool, and sign selection is visual - rather than drop-down list vs. checkbox. And when trying out different themes, there's no need to keep going in to mod options.

**Comment by kianzarrin** [comment]
@aubergine10 about hotkey conflicts with Find it. one solution would be for both FindIT and TMPE to register hotkeys to UUI. that way UUI is able to resolve hotkey conflicts (active tool takes priority). you don't even need to have a UUI button.

**Comment by aubergine10** [comment]
@kianzarrin Find It has already been updated so its category hotkeys aren't global by default.

**Comment by kianzarrin** [comment]
@aubergine10 yeah I noticed but just to let you know for future. BTW TMPE stay-in-lane seems to conflict with FRT node snapping. I told klyte about this.

**Comment by aubergine10** [comment]
Now we have distinct icon themes, would be nice if chosen theme was reflected in the toolbar (eg. priority signs button shows `Yield.png`, speed limits button shows `30.png`, and parking restrictions button shows `Parking.png`).

**Comment by krzychu124** [comment]
User request (more or less) 🤣 
![image](https://user-images.githubusercontent.com/19638970/151237909-1c6769d9-4fbe-4e76-a23e-8049d3af062b.png)

I admit that normal view (not defaults) could show only sign textures from theme (maybe togglable in the options).
I'm pretty sure that current solution may confuse users, but works perfectly for debugging which is great addition. Having the old view + themes could make them really happy with the least effort from our side

**Comment by aubergine10** [comment]
There's been a few users commenting along those lines.

If we add a new mod option, maybe it should be 'off by default', something like:

```
( ) Use un-themed green signs for default speeds in normal view
```

So by default the normal view will always show themed signs regardless of whether they are default speed or custom speed override.

Maybe we could do an 11.6.5 release with that in?

**Comment by aubergine10** [comment]
Option added: 
![image](https://user-images.githubusercontent.com/1386719/151250432-c0a6a8c7-878f-40a0-a2ef-1cbfae554296.png)

Now just need to work out where in the speed limits code determines which icon to show.
